servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -677,6 +676,7 @@ boolean processGroup(ObjectStoreConnection con, Group originalGroup, final Strin
             requestSuccess = con.updateGroup(domainName, group);
         }
 
+        requestSuccess = processGroupTags(group, groupName, domainName, originalGroup, con) || requestSuccess;
```

Comment:
this is actually a wrong move. we cannot go ahead and process our group tags if the group update or insert operation failed. the original order is correct.